### PR TITLE
2.6 Feature/forms answers refactor

### PR DESCRIPTION
**Pull request by btry**
make code more readable

### CODE REVIEW

File: inc/form.class.php [hunk 1]
```diff
@@ -1071,18 +1071,18 @@ public function saveForm() {
       foreach ($found_questions as $id => $fields) {
          // If field was not post, it's value is empty
          if (isset($_POST['formcreator_field_' . $id])) {
-            $datas[$id] = is_array($_POST['formcreator_field_' . $id])
+            $datas['formcreator_field_' . $id] = is_array($_POST['formcreator_field_' . $id])
```
Comment (btry) [inline inc/form.class.php]: this change and othrer ones in this file are required to ensure that call to PluginFormcreatorForms_answers::saveAnswers() receive always the same array structure in its first argument.

File: inc/form_answer.class.php [hunk 2]
```diff
@@ -508,15 +508,8 @@ public function saveAnswers($datas) {
...
-                FROM glpi_plugin_formcreator_questions q
-                LEFT JOIN glpi_plugin_formcreator_sections s
-                  ON s.`id` = q.`plugin_formcreator_sections_id`
-                LEFT JOIN `glpi_plugin_formcreator_answers` AS a
-                  ON a.`plugin_formcreator_forms_answers_id` = $formanswers_id
-                  AND a.`plugin_formcreator_question_id` = q.`id`
-                WHERE s.`plugin_formcreator_forms_id` = {$datas['formcreator_form']}";
-      $result = $DB->query($query);
+      $question = new PluginFormcreatorQuestion();
+      $questions = $question->getQuestionsFromForm($datas['formcreator_form']);
```
Comment (btry) [inline inc/form_answer.class.php]: get questions of the form with a dedicated method. fields from glpi_plugin_formcreator_answers are retrieved later in the code

**Comment by btry** [comment]
the problem was in this pr, not develop branch.

Fixed

**Comment by orthagh** [comment]
OK for me

**Pull request merged**